Kickstarter-iOS/Features/FloatingTabBar/FloatingTabBar.swift

```diff
@@ -0,0 +1,208 @@
+    view.isUserInteractionEnabled = false
+    return view
+  }()
+
+  // MARK: - Init
+
+  override init(frame: CGRect) {
+    super.init(frame: frame)
+
+    self.backgroundImage = UIImage()
+    self.shadowImage = UIImage()
+    self.isTranslucent = true
+    self.backgroundColor = .clear
+
+    self.tintColor = Colors.FloatingTabBar.iconColorSelected.uiColor()
+    self.unselectedItemTintColor = Colors.FloatingTabBar.iconColorUnselected.uiColor()
+
+    let appearance = UITabBarAppearance()
+    appearance.configureWithTransparentBackground()
+    appearance.stackedLayoutAppearance.selected.iconColor = Colors.FloatingTabBar.iconColorSelected.uiColor()
+    appearance.stackedLayoutAppearance.normal.iconColor = Colors.FloatingTabBar.iconColorUnselected.uiColor()
+
+    // Hide icon labels.
+    appearance.stackedLayoutAppearance.normal.titleTextAttributes = [
+      .foregroundColor: UIColor.clear
+    ]
+    appearance.stackedLayoutAppearance.selected.titleTextAttributes = [
+      .foregroundColor: UIColor.clear
+    ]
+
+    self.standardAppearance = appearance
+    self.scrollEdgeAppearance = appearance
+
+    self.setupSubviews()
+  }
+
+  required init?(coder: NSCoder) {
+    super.init(coder: coder)
+  }
+
+  /// Move the green background when the selected item changes.
+  override var selectedItem: UITabBarItem? {
+    didSet {
+      self.updateSelection(animated: true)
+    }
+  }
+
+  override func layoutSubviews() {
+    super.layoutSubviews()
+
+    self.layoutTabs()
+
+    self.updateSelection(animated: false)
+  }
+
+  private func setupSubviews() {
+    self.tabBarBackgroundView.backgroundColor = Colors.FloatingTabBar.background.uiColor()
+    self.tabBarBackgroundView.layer.cornerRadius = Constants.tabBarCornerRadius
+    self.tabBarBackgroundView.layer.masksToBounds = false
+    self.tabBarBackgroundView.layer.shadowColor = UIColor.black.cgColor
+    self.tabBarBackgroundView.layer.shadowOpacity = Constants.tabBarShadowOpacity
+    self.tabBarBackgroundView.layer.shadowRadius = Constants.tabBarShadowRadius
+    self.tabBarBackgroundView.layer.shadowOffset = CGSize(width: 0, height: Constants.tabBarShadowOffsetY)
+
+    self.selectedTabBackgroundView.backgroundColor = Colors.FloatingTabBar.iconHighlight.uiColor()
+    self.selectedTabBackgroundView.layer.cornerRadius = Constants.selectedTabBackgroundCornerRadius
+    self.selectedTabBackgroundView.clipsToBounds = true
+
+    addSubview(self.tabBarBackgroundView)
+    addSubview(self.selectedTabBackgroundView)
+    addSubview(self.tabsStackView)
+
+    sendSubviewToBack(self.tabBarBackgroundView)
+  }
+
+  /// Centers and lays out the tabs.
+  private func layoutTabs() {
+    let tabViews = self.sortedItemViews()
+    let isEmpty = tabViews.isEmpty
+
+    self.tabBarBackgroundView.isHidden = isEmpty
+    self.selectedTabBackgroundView.isHidden = isEmpty
+    self.tabsStackView.isHidden = isEmpty
+
+    guard isEmpty == false else { return }
```

Comment:
Nit: instead of `isEmpty` and `tabViews[0].center.y`, you could call this like
```suggestion
    guard let firstTab == self.tabViews.first else { return }
    let iconsCenterY = firstTab.center.y
```